src/citrine/resources/material_template.py

```diff
@@ -115,3 +116,48 @@ class MaterialTemplateCollection(DataConceptsCollection[MaterialTemplate]):
     def get_type(cls) -> Type[MaterialTemplate]:
         """Return the resource type in the collection."""
         return MaterialTemplate
+
+    def get_specs(self, scope: str, id: str, per_page: int = 20) -> Iterator[dict]:
+        """
+        [ALPHA] Get all material specs associated with a material template.
+
+        The material template is specified by its scope and id.
+
+        :param scope: The scope used to locate the material template.
+        :param id: The unique id corresponding to `scope`.
+            The lookup will be most efficient if you use the Citrine ID (scope='id')
+            of the material template.
+        :param per_page: The number of results to return per page.
+        :return: A search result of material specs
+        """
+        path = self._get_path(ignore_dataset=True) + "/" + scope + "/" + id + "/material-specs"
+        return self.session.cursor_paged_resource(self.session.get_resource,
+                                                  path,
+                                                  per_page=per_page,
+                                                  version="v1")
+
+    # Retrieve all material runs associated with a material template
+    def get_runs(self,
+                 template_scope: str,
+                 template_id: str,
```

Comment:
Have you considered flipping which collection you put these methods on? I think it may be more intuitive to structure like
```python
my_project.material_runs.with_spec(...)
my_project.material_runs.with_template(...)
my_project.material_specs.with_template(...)
```
Opinion of @lkubie, @ventura-rivera, @Imperssonator, etc is probably more valuable here though.

Reply:
I really like the` with_...` access patterns as it parallels the other filtering access patterns. e.g.
`my_project.material_runs.filter_by_name("...")` @asantas93

Reply:
The issue with doing it that way is that the endpoints won't match the endpoints on the backend. I can certainly switch the endpoints on the backend, but probably worth discussing